src/mpid/ch4/shm/xpmem/xpmem_impl.h

```diff
@@ -20,22 +20,20 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_XPMEM_send_rts(int rank, MPIR_Comm * comm, in
                                                   void *buf, MPI_Aint data_sz,
                                                   int tag, MPIR_Request * sreq)
 {
-    MPIDI_SHM_ctrl_hdr_t ctrl_hdr;
-    MPIDI_SHM_ctrl_xpmem_send_lmt_rts_t *slmt_req_hdr = &ctrl_hdr.xpmem_slmt_rts;
+    MPIDI_SHM_ctrl_xpmem_send_lmt_rts_t am_hdr;
 
```

Comment:
Renaming `slmt_req_hdr` to `am_hdr` seems unrelated to this commit

Reply:
All these routine to send a am packet are similar, declare a local am_hdr structure, then filling the field, and then send them using lower-layer transport. I was thinking that the more we make these code similar, the easier we are able to read them and focus on the details, such as the structure fields and semantics. 

Yes, I can split them or remove it if you prefer.